hadoop-hdds/common/src/test/java/org/apache/hadoop/hdds/utils/db/cache/TestTableCacheImpl.java

```diff
@@ -93,21 +93,21 @@ public void testPartialTableCache() {
   public void testPartialTableCacheWithNotContinousEntries() throws Exception {
     int totalCount = 0;
     int insertedCount = 3000;
+
+    int cleanupCount = 1000;
+
+    ArrayList<Long> epochs = new ArrayList();
     for (long i=0; i<insertedCount; i+=2) {
+      if (cleanupCount++ < 1000) {
```

Comment:
The cleanupCount is set to 1000. Perhaps I am mistaken so I am reading this as -- cleanupCount is now 1001, and we check if it is less than 1000 ? I am still not sure why you wanted to check the non-continuous order, if it is to simulate threads, you probably need randomness?

Reply:
Thanks for catching. I need to initialize cleanupCount to zero, as I want to have 1000 entries in epoch list. As I want to check behavior of cleanup with not continous entries with cache having entries like that.

Reply:
Yes, but the source of the bug is that threads can be in inserting any entry under the current Epoch in different order. May I suggest that you create this list with a four loop and then do a simple shuffle. Then it does not matter if you are incrementing with 2 or 1. 

The point is that you need the cache entries out-of-order to verify your scenario.
Making it an even number list does not really qualify as an out-of-order list.

But it is fine; It is a minor detail; if you think this test cases addresses the issue sufficiently I am fine with going ahead with this change.

Reply:
http://commons.apache.org/proper/commons-lang/javadocs/api-release/org/apache/commons/lang3/ArrayUtils.html#shuffle-boolean:A-